### ISSUE TITLE
Receiving 404 error with http receiver

### ISSUE THREAD

**Issue by ogxd** [open-telemetry/opentelemetry-collector#5187]
Hello!

I am not sure if this is a bug or a configuration issue on my end, but after many tries, I haven't been able to receive anything through the HTTP receiver. 

I am using this rather simple receiver configuration:
```
receivers:
  otlp:
    protocols:
      http:
        endpoint: 0.0.0.0:4318
```

I tried several combinations, having the server hosted remotely on a K8S cluster or even locally via docker desktop, and using the CSharp client configured to send via HTTP, but **all I get is a 404**, even if I just try to connect to http://localhost:4318 the status code returned is 404, while I would rather expect something like a 405 or a response with some error information.

I tried with several "recent" versions of `opentelemetry-collector-contrib` (0.48.0, 0.47.0, ...) but it was all the same. Then I randomly tried the 0.25.0 and this time I could at least receive a `Not Implemented` response, which makes me think that the issue is on the server side.

For my local tries, here is the Dockerfile I was using:
```
FROM otel/opentelemetry-collector-contrib:0.48.0
COPY otel-config.yml .
CMD ["--config", "otel-config.yml"]
```

Here is the full `otel-config.yml` if that can help:
```
extensions:
  health_check:
  - endpoint: 0.0.0.0:13133

receivers:
  otlp:
    protocols:
      http:
        endpoint: 0.0.0.0:4318

processors:
  batch:
    timeout: 10s

exporters:
  logging:
    logLevel: debug

service:
  extensions: [health_check]
  pipelines:
    traces:
      receivers: [otlp]
      processors: [batch]
      exporters: [logging]
```

**Comment by bogdandrutu** [comment]
That is because we register only on paths "0.0.0.0:4318/v1/traces", "0.0.0.0:4318/v1/metrics" and "0.0.0.0:4318/v1/logs". Any other path is 404.

**Comment by ogxd** [comment]
Ah perfect I finally got it working, thank you.   
I've seen this mentioned here, however I did not see any example and I thought that maybe the opentelemetry-dotnet lib I am using would append this by itself. Given how easy it is to swap grpc with http in the yml, it felt normal to only swap the port for the endpoint I use :)     
So it's definitely not a bug, but I think it could be interesting to return something else than an empty 404 even is the endpoint is incorrect, so that the user knows that the service is there.    
**Example:**
Valid: https://api.github.com/users/ogxd -> status 200
Invalid: https://api.github.com/imnotanenpoint -> status 404 but with a reponse
